tomcat-8.5/src/org/dogtagpki/tomcat/JSSUtil.java

```diff
@@ -66,6 +61,30 @@ public JSSUtil(SSLHostConfigCertificate cert) {
         logger.debug("JSSUtil: instance created");
     }
 
+    private void init() {
+        if (engine != null) {
+            return;
+        }
+
+        try {
+            JSSContext ctx = new JSSContext(null);
+            ctx.init(null, null, null);
+            engine = ctx.createSSLEngine();
+        } catch (Exception e) {
+            // Nothing to do. Better to return a NULL list than throw an
+            // exception. SunJSSE shouldn't throw an exception with a null
+            // initialization list and nor will JSS. This shouldn't happen.
```

Comment:
I generally would disagree with this approach. Whether an exception "shouldn't"
or "won't" happen is irrelevant. When a library declares a checked exception, the
exception technically "might" happen, either under certain circumstances only or
maybe in a future version of the library. It's the caller's responsiblity to handle it
or let it propagate.

I think if we don't know how to handle a checked exception we should always
rethrow the exception and let somebody else deal with that, for example:
```
throw new RuntimeException(e);
```
and it's shorter than explaining why the exception "shouldn't" happen. :)

Theoretically if the original exception "shouldn't" happen, this code "shouldn't"
run either, so there's no harm having this code in the catch block. However, in
the rare that case it does happen, this code will allow the user to see what went
wrong. If we simply ignore the exception, the information will be lost and it will
be difficult to troubleshoot the problem.